### PR TITLE
docs(gatsby-react-router-scroll): Add documentation for scroll restoration options

### PR DESCRIPTION
**Pull request by blainekasten**
## Description

Adds `useScrollRestoration` to main exports from `gatsby`
Adds a small intro to our scroll restoration. Turns out we weren't documenting these _anywhere_.

Clubhouse ticket: https://app.clubhouse.io/gatsbyjs/story/9575/add-documentation-for-usescrollrestoration

## Related Issues

#24306

### CODE REVIEW

File: docs/docs/reach-router-and-gatsby.md [hunk 1]
```diff
@@ -20,6 +21,46 @@ With Gatsby, you will mostly be using the `<Link />` component provided by the `
 
 Besides using the [`<Link />` API](https://www.gatsbyjs.org/docs/gatsby-link/) for linking between pages Gatsby generates, you can define your own client-side routes. See the [client only paths example](https://github.com/gatsbyjs/gatsby/tree/master/examples/client-only-paths) on how to use `<Router />` from `@reach/router` to make client routes work seamlessly together with your server routes.
 
+## Scroll Restoration
+
+Gatsby will handle scroll restoration for you in most cases. However, when you render containers that have their own scroll values, those scroll positions are typically lost between page transitions. To solve that, users can use the (deprecated) `ScrollContainer` component or `useScrollRestoration` hook in their code to tell Gatsby about scroll containers that we should track and restore.
```
Comment (madalynrose) [inline docs/docs/reach-router-and-gatsby.md:26]: I'd list the recommended `useScrollRestoration` before the deprecated fallback option. Thank you for adding these, it's so helpful! I was just crawling through code trying to figure out how it works for CSR a11y improvements.

Comment (blainekasten) [reply docs/docs/reach-router-and-gatsby.md:26]: Awesome! Will do

Comment (blainekasten) [reply docs/docs/reach-router-and-gatsby.md:26]: Oooo, so I assumed this was an a11y improvement but I didn't know for sure. Should we add a reference to this from any a11y documentation?

File: docs/docs/reach-router-and-gatsby.md [hunk 2]
```diff
@@ -11,6 +11,7 @@ The main reasons Gatsby uses `@reach/router` are:
 1. Preloading. You can read more about preloading in the docs for the [Gatsby Link API](https://www.gatsbyjs.org/docs/gatsby-link/).
 2. The [routing accessibility](https://reach.tech/router/accessibility) it provides.
 3. It supports [server rendering](https://reach.tech/router/server-rendering) which helps Gatsby build routed files at build time.
+4. Scroll restoration.
```
Comment (AishaBlake) [inline docs/docs/reach-router-and-gatsby.md:14]: This should probably be a full sentence with a link to learn more, like the other list items.

File: packages/gatsby/index.d.ts [hunk 3]
```diff
@@ -41,7 +48,7 @@ export const prefetchPathname: (path: string) => void
  * export default (props: PageProps) => {
  *
  * @example
- * // When adding types for both pageContext (represended by LocaleLookUpInfo) 
+ * // When adding types for both pageContext (represended by LocaleLookUpInfo)
```
Comment (AishaBlake) [inline packages/gatsby/index.d.ts:51]:
```suggestion
 * // When adding types for both pageContext (represented by LocaleLookUpInfo)
```

Might as well fix a typo while we're in here.

File: docs/docs/reach-router-and-gatsby.md (hunk 2 above)
Comment (blainekasten) [reply docs/docs/reach-router-and-gatsby.md:14]: I like this change, but I don't really know what to link to. Maybe this? https://reacttraining.com/react-router/web/guides/scroll-restoration

Comment (AishaBlake) [reply docs/docs/reach-router-and-gatsby.md:14]: Sorry! Yes, that's what I meant by "Reach docs". That was, in no way, clear. 😳

Comment (blainekasten) [reply docs/docs/reach-router-and-gatsby.md:14]: fwiw that's react router docs. Reach doesn't have a section on scroll restoration

Comment (AishaBlake) [reply docs/docs/reach-router-and-gatsby.md]: That... is true. It _is_ the page I was thinking of, regardless. 😅

File: docs/docs/scroll-restoration.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+    return (
+        <ul style={{ height: 200, overflow: `auto` }} {...ulScrollRestoration}>
+            {countryList.map(country => <li>{country}</li>)
+        </ul>
+    );
+}
+```
+
+This is an example of using the (deprecated) ScrollContainer component with the same code.
```
Comment (AishaBlake) [inline docs/docs/scroll-restoration.md:26]:
```suggestion
This is an example of using the (deprecated) `ScrollContainer` component with the same code.
```

**Pull request merged**